### PR TITLE
feat: runtime parity for handling both ElasticSearch 6 and 7 upstreams

### PR DESCRIPTION
**Pull request by bietkul**
### What does this do / why do we need it?
This PR merges the Arc codebase (which was previously separate) for ElasticSearch v6 and ElasticSearch v7.

### What should your reviewer look out for in this PR?
Everything - it's a major refactor.

### CODE REVIEW

File: util/esclient.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+func getURL() string {
+	url := os.Getenv("ES_CLUSTER_URL")
+	if url == "" {
+		log.Fatal("Error encountered: ", fmt.Errorf("ES_CLUSTER_URL must be set in the environment variables"))
+	}
+	return url
+}
+
+// EnableTestMode enables the testing mode
+func EnableTestMode() {
```
Comment (siddharthlatest) [inline util/esclient.go]: @bietkul What is this supposed to do?

File: plugins/reindexer/service.go [hunk 2]
```diff
@@ -1,14 +0,0 @@
-package reindexer
```
Comment (siddharthlatest) [inline plugins/reindexer/service.go:1]: Why is this file removed?

File: util/esclient.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+			initClient7()
+		})
+	}
+	return client7
+}
+
+// GetClient6 returns the es6 client
+func GetClient6() *es6.Client {
+	if client6 == nil {
+		clientInit.Do(func() {
```
Comment (bietkul) [inline util/esclient.go]: Remove it.

File: util/esclient.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+			initClient6()
+		})
+	}
+	return client6
+}
+
+// GetVersion returns the es version
+func GetVersion() int {
+	if isTestingMode {
```
Comment (bietkul) [inline util/esclient.go]: Remove it.

File: plugins/auth/dao.go [hunk 5]
```diff
@@ -12,49 +11,34 @@ import (
 	"github.com/appbaseio/arc/model/permission"
 	"github.com/appbaseio/arc/model/user"
 	"github.com/appbaseio/arc/util"
-	"github.com/olivere/elastic/v7"
 )
 
-type elasticsearch struct {
-	url                             string
+type elasticSearch struct {
```
Comment (bietkul) [inline plugins/auth/dao.go]: Check on the name(camel case)

File: util/retrier.go [hunk 6]
```diff
@@ -7,18 +7,18 @@ import (
...
 type Retrier struct {
-	backoff elastic.Backoff
+	backoff es7.Backoff
 }
 
 // NewRetrier returns a new retrier with exponential backoff strategy.
 func NewRetrier() *Retrier {
 	return &Retrier{
-		elastic.NewExponentialBackoff(10*time.Millisecond, 8*time.Second),
+		es7.NewExponentialBackoff(10*time.Millisecond, 8*time.Millisecond),
```
Comment (bietkul) [inline util/retrier.go]: Change it to `time.Second`

File: util/esclient.go [hunk 7]
```diff
@@ -24,30 +23,22 @@ var (
 // GetClient7 returns the es7 client
 func GetClient7() *es7.Client {
 	// initialize the client if not present
-	if client7 == nil {
-		clientInit.Do(func() {
-			initClient7()
-		})
-	}
+	clientInit.Do(func() {
```
Comment (siddharthlatest) [inline util/esclient.go]: @bietkul You removed the nil check instead of removing the `clientInit.Do(..)` call.

**Comment by bietkul** [comment]
@siddharthlatest Fixed it.

**Review by siddharthlatest — APPROVED** [review]
lgtm (y)
